### PR TITLE
use BASH_SOURCE to determine KUBE_ROOT in update-bindata.sh

### LINKED ISSUE #30621: hack/update-bindata.sh sets KUBE_ROOT default value relative to test/e2e/generated/
I tried to launch hack/update-bindata.sh from root directory of kubernetes repo and I got following error:

`hack/update-bindata.sh: line 26: ../../..//cluster/lib/logging.sh: No such file or directory`

After looking the code, I understood that update-bindata.sh script should be launched only from test/e2e/generated/ directory or I should specify KUBE_ROOT variable by myself. It is not user-friendly at all and it would be nice to align all pathes to `${BASH_SOURCE[0]}`, so users will be able to launch this script from whatever they want path.

### PR DESCRIPTION
**Pull request by andreykurilin**
Fixes #30621

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30622)

### CODE REVIEW

**Comment by eparis** [comment]
Don't we frown on readlink before of mac OS? I believe you need to source `lib/init.sh` and use `readlinkdashf`.   However, the idiomatic was to do this in kube is just to do:

`KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../../..`

like all of the other things in `hack`.

**Comment by andreykurilin** [comment]
@eparis imo, we should transform path to real form, since we prints paths end-users and relative pathes are not looks well

**Comment by andreykurilin** [comment]
@eparis thanks for readlinkdashf function. Reused it now

File: hack/update-bindata.sh [hunk 1]
```diff
@@ -19,8 +19,8 @@ set -o pipefail
 set -o nounset
 
 if [[ -z "${KUBE_ROOT:-}" ]]; then
-	# Relative to test/e2e/generated/
-	KUBE_ROOT="../../../"
+    source $(dirname "${BASH_SOURCE[0]}")"/lib/init.sh"
+    KUBE_ROOT=$(kube::readlinkdashf $KUBE_ROOT)
```
Comment (eparis) [inline hack/update-bindata.sh]: but $KUBE_ROOT is `-z`...

Comment (andreykurilin) [inline hack/update-bindata.sh]: init.sh sets KUBE_ROOT by itself, so after `source` is executed, KUBE_ROOT should be available

**Pull request closed without merging**